### PR TITLE
Adjusting currentResource - Stops re-rendering of Header

### PR DESCRIPTION
**Pull request by jcontonio**

### CODE REVIEW

File: web/src/client/js/components/Project/ProjectContainer.js [hunk 1]
```diff
@@ -3,18 +3,19 @@ import PropTypes from 'prop-types'
 import { withRouter } from 'react-router-dom'
 
 import useDataService from '../../hooks/useDataService'
-import dataMapper from '../../libs/dataMapper'
+// import dataMapper from '../../libs/dataMapper'
```
Comment (dheniges) [inline web/src/client/js/components/Project/ProjectContainer.js]: hashtag deleteme

File: web/src/client/js/components/Header/Header.js [hunk 2]
```diff
@@ -1,58 +1,33 @@
...
+    </div>
   )
 }
 
 Header.propTypes = {
-  match: PropTypes.object.isRequired,
-  currentUser: PropTypes.object
+  brand: PropTypes.object,
+  match: PropTypes.object,
+  history: PropTypes.object
```
Comment (dheniges) [inline web/src/client/js/components/Header/Header.js]: history is no longer a used props on this component

File: web/src/client/js/libs/currentResource.js [hunk 3]
```diff
@@ -1,38 +1,43 @@
...
-        fetchAction: () => {},
-        getLoadingStatusFromState: () => {},
-        getDataFromState: () => {}
-      }
-    },
-    idName: null
+function returnNull() {
+  return {
+    fetchAction: () => {},
+    getLoadingStatusFromState: () => {},
```
Comment (dheniges) [inline web/src/client/js/libs/currentResource.js]:
```suggestion
    getLoadingStatusFromState: () => false,
```
The current implementation has a bug where once the resource is set, if it moves to an undefined resource the previous resource will never be unset. Returning false here tells useDataService the resource is loaded and will then set the appropriate (undefined in this case) data and the component using the hook will update accordingly

**Review by dheniges — CHANGES_REQUESTED** [review]
A few small changes, and see my suggestion (I tested it locally, it works).

I don't love the reliance on the history object because it's mutable and there are no guarantees the component will update when it's mutated, or update how you'd expect it to. But it seems to work 🤷‍♀️

**Update**
After further testing, the location object exposed by react router is immutable, so we should replace reading the pathname with `location.pathname` to avoid future state bugs

File: web/src/client/js/libs/currentResource.js (hunk 3 above)
Comment (jcontonio) [reply web/src/client/js/libs/currentResource.js]: Nice

File: web/src/client/js/components/Navigator/NavigatorContainer.js [hunk 4]
```diff
@@ -12,9 +12,11 @@ import DropdownSelectComponent from 'Components/DropdownSelect/DropdownSelectCom
 
 import './Navigator.scss'
 
-const NavigatorContainer = ({ match, history }) => {
+const NavigatorContainer = ({ history }) => {
   const { data: projects } = useDataService(dataMapper.projects.list())
-  const { data: project } = useDataService(currentResource(match), [match])
+  const { data: project } = useDataService(
+    currentResource('project', history.location.pathname), [history.location.pathname]
```
Comment (dheniges) [inline web/src/client/js/components/Navigator/NavigatorContainer.js]: still have some history action going on here

**Review by dheniges — CHANGES_REQUESTED** [review]
Navigator's still pulling history, otherwise looks good

**Review by dheniges — APPROVED** [review]

**Pull request merged**